lib/acme/client/resources/challenges.rb

```diff
@@ -4,18 +4,14 @@ module Acme::Client::Resources::Challenges
   require 'acme/client/resources/challenges/base'
   require 'acme/client/resources/challenges/http01'
   require 'acme/client/resources/challenges/dns01'
+  require 'acme/client/resources/challenges/unsupported_challenge'
 
   CHALLENGE_TYPES = {
     'http-01' => Acme::Client::Resources::Challenges::HTTP01,
     'dns-01' => Acme::Client::Resources::Challenges::DNS01
   }
 
   def self.new(client, type:, **arguments)
-    klass = CHALLENGE_TYPES[type]
-    if klass
-      klass.new(client, **arguments)
-    else
-      { type: type }.merge(arguments)
-    end
+    (CHALLENGE_TYPES[type] || Unsupported).new(client, **arguments)
```

Comment:
Doesn't seem related to post-as-get. What the use case for this?

Reply:
This is to gracefully handle receiving a TLS-ALPN challenge.  Basically this will fall back to a generic unsupported challenge type.

Reply:
Fair enough